Training/interface/DataLoader_main.h

```diff
@@ -204,23 +204,23 @@ class DataLoader {
       std::shared_ptr<TH2D> target_th2d = std::shared_ptr<TH2D>(dynamic_cast<TH2D*>(file_target->Get("eta_pt_hist_tau")));
       if (!target_th2d) throw std::runtime_error("Target histogram could not be loaded");
 
-      for( auto const& [tau_type, tau_name] : tau_types_names)
-      { if (tau_type != 8){ // No histogram for data samples
-        std::shared_ptr<TH2D> input_th2d  = std::shared_ptr<TH2D>(dynamic_cast<TH2D*>(file_input ->Get(("eta_pt_hist_"+tau_name).c_str())));
-        if (!input_th2d) throw std::runtime_error("Input histogram could not be loaded for tau type "+tau_name);
-        target_histogram.th2d_add(*(target_th2d.get()));
-        input_histogram .th2d_add(*(input_th2d .get()));
-
-        target_histogram.divide(input_histogram);
-        hist_weights[tau_type] = target_histogram.get_weights_th2d(
-            ("w_1_"+tau_name).c_str(),
-            ("w_1_"+tau_name).c_str()
-        );
-        if (debug) hist_weights[tau_type]->SaveAs(("Temp_"+tau_name+".root").c_str()); // It's required that all bins are filled in these histograms; save them to check incase binning is too fine and some bins are empty
-
-        target_histogram.reset();
-        input_histogram .reset();
-      } 
+      for( auto const& [tau_type, tau_name] : tau_types_names){
+          if (tau_name != "data"){ // No histogram for data samples
```

Comment:
here and in similar places below: don't use strings (string operation are much slower than integer comparison) -> use enum values